src/app/content/search/hooks/index.ts

```diff
@@ -146,10 +146,40 @@ export const syncSearch: RouteHookBody<typeof content> = (services) => async() =
   }
 };
 
+export const openSearchInSidebarHook: ActionHookBody<typeof openSearchInSidebar> = (services) => () => {
+  // Restore search state when opening sidebar
+  const state = services.getState();
+  const previousState = select.previousState(state);
+  const hash = selectNavigation.hash(state);
+  const systemQueryParams = selectNavigation.systemQueryParameters(state);
+  const { query, selectedResult } = previousState;
+
+  if (!query) {
+    return;
+  }
+
+  const options = selectedResult
+    ? {
+      hash: selectedResult.result.source.elementId,
+      search: queryString.stringify({
+        query,
+        target: JSON.stringify({ type: 'search', index: selectedResult.highlight }),
+        ...systemQueryParams,
+      }),
+    }
+    : { search: queryString.stringify({ query }) };
+
+  services.history.replace({
+    hash,
+    ...options,
+  });
+};
```

Comment:
@bethshook Curious what you think about this approach for restoring the last search state?